### DISCUSSION TITLE
Working with multiple schema (databases)

### DISCUSSION THREAD

**Discussion by nmehitabel** [komapper/komapper#912] [Q&A] [answered]
Is it possible to work with a (legacy mysql) server that has several distinct schemas in one komapper project.
My first use case may be reasonably simple in that I just want to extract data from single tables for onward processing .e.g.
`select id, name, date_registered from users.user_info` from users db
`select id,user_id,order_date from orders.order` from orders db
Is it feasible to generate the entities belonging to generate databases / schema in the same project?

If the above is possible is it then feasible to join across schemas / databases?

Thanks.

**Comment by nakamura-to** [comment, marked as answer]
> Is it possible to work with a (legacy mysql) server that has several distinct schemas in one komapper project.

Yes. You can define entity classes as follows:

```kotlin
@KomapperEntity
@KomapperTable(schema = "users")
data calss UserInfo(...)

@KomapperEntity
@KomapperTable(schema = "orders", alwaysQuote = true)
data calss Order(...)
```

> Is it feasible to generate the entities belonging to generate databases / schema in the same project?

Yes, [Komapper's Gradle plugin](https://www.komapper.org/docs/reference/gradle-plugin/) supports it.
In build.gradle.kts you can configure for each schema:

```kotlin
komapper {
    generators {
        register("users") {
            // configure for users schema
        }
        register("orders") {
            // configure for orders schema
        }
    }
}
```

> If the above is possible is it then feasible to join across schemas / databases?

Yes, it is possible if MySQL supports it.

**Reply by nmehitabel** [reply]
Perfect. Thanks very much.
